src/chem/viirs_aod2ioda.py

```diff
@@ -9,189 +10,236 @@
 import sys
 import argparse
 import netCDF4 as nc
+import glob
 import numpy as np
-from datetime import datetime, timedelta
+from datetime import datetime, date, timedelta
+import os
 from pathlib import Path
-
+sys.stdout.flush()
 IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
 if not IODA_CONV_PATH.is_dir():
     IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
 sys.path.append(str(IODA_CONV_PATH.resolve()))
 
-import ioda_conv_ncio as iconv
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
 from orddicts import DefaultOrderedDict
 
-vName = {
-    'A': "aerosol_optical_depth_4",
-}
-
 locationKeyList = [
     ("latitude", "float"),
     ("longitude", "float"),
     ("datetime", "string")
 ]
 
-AttrData = {}
 
+obsvars = {
+    'A': "aerosol_optical_depth_4",
+}
 
-class AOD(object):
+AttrData = {
+    'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
 
-    def __init__(self, filename, method, mask, thin, writer):
-        self.filename = filename
-        self.method = method
+
+DimDict = {
+}
+
+VarDims = {
+    'aerosol_optical_depth_4': ['nlocs']
+}
+
+
+class AOD(object):
+    def __init__(self, filenames, obs_time, method, mask, thin):
+        self.filenames = filenames
+        self.obs_time = obs_time
```

Comment:
This is never used?

Reply:
I will take them out, if not needed.

Reply:
I'm not sure I understand why these lines (53-56) are not used.  Or do you just mean line 56?

Reply:
I'm talking about line 56.